agent/vendor/github.com/aws/amazon-ecs-agent/ecs-agent/api/errors/error_messages.go

```diff
@@ -0,0 +1,252 @@
+			RawError: err,
+		}, nil
+	}
+	return DockerError{RawError: err}, errors.New(ErrorMessageDoesNotMatch)
+}
+
+func parseNetworkConfigurationError(err string) (DockerError, error) {
+	matched, _ := regexp.MatchString(PatternNetworkConfigurationError, err)
+	if matched {
+		return DockerError{
+			Type:     NetworkConfigurationError,
+			RawError: err,
+		}, nil
+	}
+	return DockerError{RawError: err}, errors.New(ErrorMessageDoesNotMatch)
+}
+
+func AugmentErrMsg(namedErr NamedError, ctx ErrorContext) NamedError {
+
+	constructibleErr, ok := namedErr.(ConstructibleNamedError)
+	if !ok { // If namedErr is not a ConstructibleNamedError, return as-is.
+		return namedErr
+	}
+
+	// Augment the error message.
+	augmentedMsg := AugmentMessage(namedErr.Error(), ctx)
+	// Reconstruct new NamedError.
+	newError := constructibleErr.Constructor()(augmentedMsg)
+	return newError
+}
+
+// Extend error messages with extra useful information.
+// Tries to find known cases by matching `errMsg` against implemented parsers for known errors.
+// If the error message pattern is unknown, returns the original error message.
+//
+// Possible failure scenarios:
+// 1. Missing parser implementation.
+//
+// This can only occur if a new ErrorType was added incorrectly due to implementation oversight.
+//
+// Currently, the function is set up in a safe manner - all failures are ignored, and
+// the original error message string is returned.
+func AugmentMessage(errMsg string, ctx ErrorContext) string {
+	var errorData DockerError
+	var err error
+
+	// Try parsing each error type until a match is found.
+	for _, parser := range errorParsers {
+		errorData, err = parser(errMsg)
+		if err == nil {
+			// Parser found a match, break out of loop
+			break
+		}
+	}
+
+	// none of the parsers match - return original untouched error message
+	if err != nil {
+		return errMsg
+	}
+
+	// lookup corresponding error formatter function. If not found - return original.
+	formattedErrorMessage, ok := errorMessageFunctions[errorData.Type]
+	if !ok {
+		return errMsg
+	}
+
+	return formattedErrorMessage(errorData, ctx)
+}
+
+// Generates error message for MissingECRBatchGetImage and ECRImageDoesNotExist errors.
+func formatMissingImageOrPullImageError(errorData DockerError, ctx ErrorContext) string {
+	rawError := errorData.RawError
+	roleName := ""
+
+	if ctx.ExecRole != "" {
+		roleName = "'" + ctx.ExecRole + "' "
+	}
+
+	formattedMessage := fmt.Sprintf(
+		"Check if image exists and role %shas permissions to pull images from Amazon ECR. %s",
+		roleName, rawError)
+
+	return formattedMessage
+}
+
+// Generates error message for MissingECRBatchGetImage and ECRImageDoesNotExist errors.
+func formatNetworkConfigurationError(errorData DockerError, ctx ErrorContext) string {
+	rawError := errorData.RawError
+	entity := ""
+
+	// Valid network modes
+	isHostNetworkMode := map[string]bool{
+		"bridge":  true,
+		"host":    true,
+		"none":    true,
+		"default": true,
```

Comment:
I don't think "default" is a "real" network mode value. 

https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RegisterTaskDefinition.html#ECS-RegisterTaskDefinition-request-networkMode 

the "default" mode is just when network mode is not set, so maybe we should only check for `awsvpc` here, and consider everything else "host network mode"

Reply:
sounds good